types/reply.d.ts

```diff
@@ -18,13 +19,13 @@ export interface FastifyReply<
   context: FastifyContext<ContextConfig>;
   log: FastifyLoggerInstance;
   request: FastifyRequest<RequestGeneric, RawServer, RawRequest>;
-  code(statusCode: number): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
-  status(statusCode: number): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
+  code(statusCode: number): FastifyReply<ReplyPayload, RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
+  status(statusCode: number): FastifyReply<ReplyPayload, RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
   statusCode: number;
   sent: boolean;
-  send<T>(payload?: T): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
-  header(key: string, value: any): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
-  headers(values: {[key: string]: any}): FastifyReply<RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
+  send(payload?: ReplyPayload): FastifyReply<ReplyPayload, RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
```

Comment:
```suggestion
  send<T = ReplyPayload>(payload?: T): FastifyReply<T, RawServer, RawRequest, RawReply, RequestGeneric, ContextConfig>;
```

This should support the existing api of specifying the send type at the call level rather than the handler level.